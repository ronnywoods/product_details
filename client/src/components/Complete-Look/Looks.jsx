import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Looks extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sizeShirt: 'SELECT SIZE',
      sizeJacket: 'SELECT SIZE',
      sizePant: 'SELECT SIZE',
      bagClassShirt: 'disabled',
      bagClassJacket: 'disabled',
      bagClassPant: 'disabled',
      purchaseShirt: true,
      purchasePant: true,
      purchaseJacket: true,
      shirt: false,
      pant: false,
      jacket: false,
    };
    this.selectSizeShirt = this.selectSizeShirt.bind(this);
    this.selectSizeJacket = this.selectSizeJacket.bind(this);
    this.selectSizePant = this.selectSizePant.bind(this);
    this.selectShirt = this.selectShirt.bind(this);
    this.selectJacket = this.selectJacket.bind(this);
    this.selectPant = this.selectPant.bind(this);
  }

  selectSizeShirt(e) {
    this.setState({ sizeShirt: e.target.textContent });
    this.setState({ bagClassShirt: 'enabled', purchaseShirt: false });
    console.log(this.state.purchaseShirt);
  }

  selectSizeJacket(e) {
    this.setState({ sizeJacket: e.target.textContent });
    this.setState({ bagClassJacket: 'enabled' });
    this.setState({ purchaseJacket: false });
  }

  selectSizePant(e) {
    this.setState({ sizePant: e.target.textContent });
    this.setState({ bagClassPant: 'enabled' });
    this.setState({ purchasePant: false });
  }

  selectShirt() {
    this.setState({ shirt: !this.state.shirt });
    this.setState({ pant: false });
    this.setState({ jacket: false });
  }

  selectPant() {
    this.setState({ pant: !this.state.pant });
    this.setState({ shirt: false, jacket: false });
  }

  selectJacket() {
    this.setState({ jacket: !this.state.jacket });
    this.setState({ shirt: false, pant: false });
  }

  render() {
    const cardShirt = this.state.shirt ? 'look-card bottom-bold' : 'look-card';
    const cardPant = this.state.pant ? 'look-card bottom-bold' : 'look-card';
    const cardJacket = this.state.jacket ? 'look-card bottom-bold' : 'look-card';
    return (
      <div className="looks">
        <div className="looks-title">
          <h4> COMPLETE THE LOOK </h4>
        </div>
        <div className="column">
          <div className="looks-pics">
            <div className={cardShirt} id="shirtCard" onClick={this.selectShirt} data-toggle="collapse" data-target="#shirtView" role="button" aria-expanded="false" aria-controls="shirtView">
              <h6>
                <img src={this.props.looks.shirt_url} />
                <br />
                <br />
                ${this.props.looks.shirt_price}
              </h6>
            </div>
            <div className={cardJacket} id="jacketCard" onClick={this.selectJacket} data-toggle="collapse" data-target="#jacketView" role="button" aria-expanded="false" aria-controls="jacketView">
              <h6>
                <img src={this.props.looks.jacket_url} />
                <br />
                <br />
                ${this.props.looks.jacket_price}
              </h6>
            </div>
            <div className={cardPant} id="pantCard" onClick={this.selectPant} data-toggle="collapse" data-target="#pantView" role="button" aria-expanded="false" aria-controls="pantView">
              <h6>
                <img src={this.props.looks.pant_url} />
                <br />
                <br />
                ${this.props.looks.pant_price}
              </h6>
            </div>
          </div>
          <div className="collapse" id="shirtView" data-parent=".looks" aria-labelledby="shirtCard">
            <div className="container">
              <div className="row">
                <div className="col-sm look-img">
                  <img src={this.props.looks.shirt_url} height="180" width="180" />
                </div>
                <div className="col-sm look-desc">
                  <h5 className="look-clothing-name">{this.props.looks.shirt_name}</h5>
                  <h6 className="look-clothing-price">${this.props.looks.shirt_price}</h6>
                  <div className="dropdown dropdown-complete-look">
                    <button className="btn select-size-btn" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                      <span className="float-left look-size">{this.state.sizeShirt}</span>
                      <i className="arrow down" />
                    </button>
                    <div className="dropdown-menu select-size-dropdown" aria-labelledby="dropdownMenuButton">
                      <div className="select-size-menu">
                        <div className="select-size" onClick={this.selectSizeShirt}>S</div>
                        <div className="select-size" onClick={this.selectSizeShirt}>M</div>
                        <div className="select-size" onClick={this.selectSizeShirt}>L</div>
                        <div className="select-size" onClick={this.selectSizeShirt}>XL</div>
                        <div className="select-size" onClick={this.selectSizeShirt}>2XL</div>
                      </div>
                    </div>
                    <br />
                    <button className={this.state.bagClassShirt} disabled={this.state.purchaseShirt} id="add-to-bag" onClick={this.props.add}>
                      <span className="bag-text">ADD TO BAG</span>
                      <span id="arrow-right">&#8594;</span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="collapse" id="jacketView" data-parent=".looks" aria-labelledby="jacketCard">
            <div className="container">
              <div className="row">
                <div className="col-sm look-img">
                  <img src={this.props.looks.jacket_url} height="180" width="180" />
                </div>
                <div className="col-sm look-desc">
                  <h5 className="look-clothing-name">{this.props.looks.jacket_name}</h5>
                  <h6 className="look-clothing-price">${this.props.looks.jacket_price}</h6>
                  <div className="dropdown">
                    <button className="btn select-size-btn" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                      <span className="float-left look-size">{this.state.sizeJacket}</span>
                      <i className="arrow down" />
                    </button>
                    <div className="dropdown-menu select-size-dropdown" aria-labelledby="dropdownMenuButton">
                      <div className="select-size-menu">
                        <div className="select-size" onClick={this.selectSizeJacket}>S</div>
                        <div className="select-size" onClick={this.selectSizeJacket}>M</div>
                        <div className="select-size" onClick={this.selectSizeJacket}>L</div>
                        <div className="select-size" onClick={this.selectSizeJacket}>XL</div>
                        <div className="select-size" onClick={this.selectSizeJacket}>2XL</div>
                      </div>
                    </div>
                    <br />
                    <button className={this.state.bagClassJacket} disabled={this.state.purchaseJacket} id="add-to-bag" onClick={this.props.add}>
                      <span className="bag-text">ADD TO BAG</span>
                      <span id="arrow-right">&#8594;</span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="collapse" id="pantView" data-parent=".looks" aria-labelledby="pantCard">
            <div className="container">
              <div className="row">
                <div className="col-sm look-img">
                  <img src={this.props.looks.pant_url} height="180" width="180" />
                </div>
                <div className="col-sm look-desc">
                  <h5 className="look-clothing-name">{this.props.looks.pant_name}</h5>
                  <h6 className="look-clothing-price">${this.props.looks.pant_price}</h6>
                  <div className="dropdown">
                    <button className="btn select-size-btn" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                      <span className="float-left look-size">{this.state.sizePant}</span>
                      <i className="arrow down" />
                    </button>
                    <div className="dropdown-menu select-size-dropdown" aria-labelledby="dropdownMenuButton">
                      <div className="select-size-menu">
                        <div className="select-size" onClick={this.selectSizePant}>S</div>
                        <div className="select-size" onClick={this.selectSizePant}>M</div>
                        <div className="select-size" onClick={this.selectSizePant}>L</div>
                        <div className="select-size" onClick={this.selectSizePant}>XL</div>
                        <div className="select-size" onClick={this.selectSizePant}>2XL</div>
                      </div>
                    </div>
                    <br />
                    <button className={this.state.bagClassPant} disabled={this.state.purchasePant} id="add-to-bag" onClick={this.props.add}>
                      <span className="bag-text">ADD TO BAG</span>
                      <span id="arrow-right">&#8594;</span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Looks.propTypes = {
  looks: PropTypes.object,
};

export default Looks;
