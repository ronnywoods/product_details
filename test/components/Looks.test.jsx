import React from 'react';
import { shallow } from 'enzyme';

import Looks from '../../client/src/components/Complete-Look/Looks';

describe('Looks Components', () => {
  const sample = {
    id: 1,
    pant_name: 'TREFOIL PANTS',
    pant_url: 'https://s3-us-west-1.amazonaws.com/adidas-shoe/pant1.jpg',
    pant_price: 75,
    shirt_name: 'OUTLINE TREFOIL TEE',
    shirt_url: 'https://s3-us-west-1.amazonaws.com/adidas-shoe/shirt4.jpg',
    shirt_price: 40,
    jacket_name: 'SST WINDBREAKER',
    jacket_url: 'https://s3-us-west-1.amazonaws.com/adidas-shoe/jacket1.jpg',
    jacket_price: 80,
  };

  const component = shallow(<Looks looks={sample} />);

  it('should display like snapshot', () => {
    expect(component).toMatchSnapshot();
  })

  it('should display shirt when picture is clicked', () => {
    component.find('#shirtCard').simulate('click');
    expect(component.find('#shirtCard').hasClass('collapsed')).toBe(false);
    expect(component.find('#shirtCard')).toHaveClassName('bottom-bold');
  });

  it('should toggle between pant display properly on click', () => {
    expect(component.find('#pantCard').hasClass('bottom-bold')).toBe(false);

    component.find('#pantCard').simulate('click');
    expect(component.find('#pantCard').hasClass('collapsed')).toBe(false);
    expect(component.find('#pantCard')).toHaveClassName('bottom-bold');

    component.find('#pantCard').simulate('click');
    expect(component.find('#pantCard').hasClass('bottom-bold')).toBe(false);
  });

  it('should not allow to add to cart when size is not selected', () => {
    component.find('#jacketCard').simulate('click');
    expect(component.find('span.look-size').at(1).text()).toBe('SELECT SIZE');
    expect(component.find('button').at(1)).toHaveClassName('disabled');
  });

  xit('should display correct size when size is selected', () => {
    component.find('#jacketCard').simulate('click');
    component.find('.dropdown').at(1).simulate('click');
    component.find('.select-size').at(5).simulate('click');
    component.find('.select-size').at(5).simulate('click', { target: { textContent: 'S' } });
    expect(component.find('span.look-size').at(1).text()).toBe('S');
  });

  xit('should allow to add to cart after size has been selected', () => {
    component.find('#jacketCard').simulate('click');
    component.find('.dropdown').at(1).simulate('click');
    component.find('.select-size').at(5).simulate('click');
    component.find('.select-size').at(5).simulate('click', { target: { textContent: 'S' } });
    expect(component.find('span.look-size').at(1).text()).toBe('S');
    expect(component.find('.bag-text').at(1)).toHaveClassName('enabled');
  });
});
