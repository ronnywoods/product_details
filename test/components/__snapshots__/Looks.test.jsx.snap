// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Looks Components should display like snapshot 1`] = `
<div
  className="looks"
>
  <div
    className="looks-title"
  >
    <h4>
       COMPLETE THE LOOK 
    </h4>
  </div>
  <div
    className="column"
  >
    <div
      className="looks-pics"
    >
      <div
        aria-controls="shirtView"
        aria-expanded="false"
        className="look-card"
        data-target="#shirtView"
        data-toggle="collapse"
        id="shirtCard"
        onClick={[Function]}
        role="button"
      >
        <h6>
          <img
            src="https://s3-us-west-1.amazonaws.com/adidas-shoe/shirt4.jpg"
          />
          <br />
          <br />
          $
          40
        </h6>
      </div>
      <div
        aria-controls="jacketView"
        aria-expanded="false"
        className="look-card"
        data-target="#jacketView"
        data-toggle="collapse"
        id="jacketCard"
        onClick={[Function]}
        role="button"
      >
        <h6>
          <img
            src="https://s3-us-west-1.amazonaws.com/adidas-shoe/jacket1.jpg"
          />
          <br />
          <br />
          $
          80
        </h6>
      </div>
      <div
        aria-controls="pantView"
        aria-expanded="false"
        className="look-card"
        data-target="#pantView"
        data-toggle="collapse"
        id="pantCard"
        onClick={[Function]}
        role="button"
      >
        <h6>
          <img
            src="https://s3-us-west-1.amazonaws.com/adidas-shoe/pant1.jpg"
          />
          <br />
          <br />
          $
          75
        </h6>
      </div>
    </div>
    <div
      aria-labelledby="shirtCard"
      className="collapse"
      data-parent=".looks"
      id="shirtView"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-sm look-img"
          >
            <img
              height="180"
              src="https://s3-us-west-1.amazonaws.com/adidas-shoe/shirt4.jpg"
              width="180"
            />
          </div>
          <div
            className="col-sm look-desc"
          >
            <h5
              className="look-clothing-name"
            >
              OUTLINE TREFOIL TEE
            </h5>
            <h6
              className="look-clothing-price"
            >
              $
              40
            </h6>
            <div
              className="dropdown dropdown-complete-look"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                className="btn select-size-btn"
                data-toggle="dropdown"
                id="dropdownMenuButton"
                type="button"
              >
                <span
                  className="float-left look-size"
                >
                  SELECT SIZE
                </span>
                <i
                  className="arrow down"
                />
              </button>
              <div
                aria-labelledby="dropdownMenuButton"
                className="dropdown-menu select-size-dropdown"
              >
                <div
                  className="select-size-menu"
                >
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    S
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    M
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    L
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    XL
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    2XL
                  </div>
                </div>
              </div>
              <br />
              <button
                className="disabled"
                disabled={true}
                id="add-to-bag"
              >
                <span
                  className="bag-text"
                >
                  ADD TO BAG
                </span>
                <span
                  id="arrow-right"
                >
                  →
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="jacketCard"
      className="collapse"
      data-parent=".looks"
      id="jacketView"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-sm look-img"
          >
            <img
              height="180"
              src="https://s3-us-west-1.amazonaws.com/adidas-shoe/jacket1.jpg"
              width="180"
            />
          </div>
          <div
            className="col-sm look-desc"
          >
            <h5
              className="look-clothing-name"
            >
              SST WINDBREAKER
            </h5>
            <h6
              className="look-clothing-price"
            >
              $
              80
            </h6>
            <div
              className="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                className="btn select-size-btn"
                data-toggle="dropdown"
                id="dropdownMenuButton"
                type="button"
              >
                <span
                  className="float-left look-size"
                >
                  SELECT SIZE
                </span>
                <i
                  className="arrow down"
                />
              </button>
              <div
                aria-labelledby="dropdownMenuButton"
                className="dropdown-menu select-size-dropdown"
              >
                <div
                  className="select-size-menu"
                >
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    S
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    M
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    L
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    XL
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    2XL
                  </div>
                </div>
              </div>
              <br />
              <button
                className="disabled"
                disabled={true}
                id="add-to-bag"
              >
                <span
                  className="bag-text"
                >
                  ADD TO BAG
                </span>
                <span
                  id="arrow-right"
                >
                  →
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="pantCard"
      className="collapse"
      data-parent=".looks"
      id="pantView"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-sm look-img"
          >
            <img
              height="180"
              src="https://s3-us-west-1.amazonaws.com/adidas-shoe/pant1.jpg"
              width="180"
            />
          </div>
          <div
            className="col-sm look-desc"
          >
            <h5
              className="look-clothing-name"
            >
              TREFOIL PANTS
            </h5>
            <h6
              className="look-clothing-price"
            >
              $
              75
            </h6>
            <div
              className="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                className="btn select-size-btn"
                data-toggle="dropdown"
                id="dropdownMenuButton"
                type="button"
              >
                <span
                  className="float-left look-size"
                >
                  SELECT SIZE
                </span>
                <i
                  className="arrow down"
                />
              </button>
              <div
                aria-labelledby="dropdownMenuButton"
                className="dropdown-menu select-size-dropdown"
              >
                <div
                  className="select-size-menu"
                >
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    S
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    M
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    L
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    XL
                  </div>
                  <div
                    className="select-size"
                    onClick={[Function]}
                  >
                    2XL
                  </div>
                </div>
              </div>
              <br />
              <button
                className="disabled"
                disabled={true}
                id="add-to-bag"
              >
                <span
                  className="bag-text"
                >
                  ADD TO BAG
                </span>
                <span
                  id="arrow-right"
                >
                  →
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
